src/ApiStatus.ts

```diff
@@ -0,0 +1,198 @@
+import { fileBackedObject } from "./FileBackedObject";
+import { SharedSettings } from "./SharedSettings";
+import APIStatusAPI, {APIStatus} from './ApiStatusApi';
+
+import Discord = require("discord.js");
+
+// map from api name (e.g. champion-mastery-v3) to the string used in the embed field
+// only apis with issues (troubled/down) are stored
+interface ApiStates {
+    [key: string]: string;    
+}
+
+interface StatusEmbedState {
+    api: ApiStates;
+    onFire: boolean;
+    allApisOK: boolean;
+    allApisIssues: boolean;
+}
+
+export default class ApiStatus {
+    private bot: Discord.Client;
+    private sharedSettings: SharedSettings;
+    private command: string;
+    private apiStatusAPI: APIStatusAPI
+
+    private lastCheckTime: number;
+    private currentStatus: StatusEmbedState;
+
+    constructor(bot: Discord.Client, sharedSettings: SharedSettings) {
+        console.log("Requested API Status extension..");
+        this.bot = bot;
+        this.command = sharedSettings.apiStatus.command;
+
+        this.sharedSettings = sharedSettings;
+        this.lastCheckTime = 0;
+        this.apiStatusAPI = new APIStatusAPI(this.sharedSettings.apiStatus.statusUrl, this.sharedSettings.apiStatus.checkInterval);
+
+        this.bot.on("ready", this.onBot.bind(this));
+        this.bot.on("message", this.onInfo.bind(this));
+    }
+
+    onBot() {
+        console.log("API Status extension loaded.");
+    }
+
+    async onInfo(message: Discord.Message) {
+        if (message.author.bot) return;
+
+        const content = message.cleanContent;
+        if (!this.isApiStatusCommand(content))
+            return;
+
+        let apiStatus = await this.getApiStatus();
+
+        let fields: Array<{name: string, value: string, inline: boolean}> = [];
+        for (let api in apiStatus.api){ 
+            fields.push({name: api, value: apiStatus.api[api], inline: true})
+        }
+
+        if (!apiStatus.allApisIssues) {
+            fields.push({
+                "name": apiStatus.allApisOK ? "All APIs" : "All other APIs",
+                "value": ":white_check_mark:",
+                "inline": true
+            })
+        }
+
+        // fixes formatting with empty fields
+        while (fields.length % 3 != 0) {
+            fields.push({
+              "name": "\u200b",
+              "value": "\u200b",
+              "inline": true
+            })
+        }
+
+        let embedContent: any = {
+            color: 0xe74c3c,
+            author: {
+                icon_url: "http://ddragon.leagueoflegends.com/cdn/7.20.2/img/champion/Heimerdinger.png",
+                name: "API Status",
+                url: "https://developer.riotgames.com/api-status/"
+            },
+            fields: fields
+        }
+
+        if (apiStatus.onFire) {
+            embedContent.image = { url: this.pickRandomOnFireImage() };
+        }
+
+        message.channel.send({embed: embedContent});
+    }
```

Comment:
Bunch of missing semi-colons and `let`s instead of `const`s in this function too